server/controllers/userPostController.js

```diff
@@ -0,0 +1,127 @@
+const jwt = require('jsonwebtoken');
+const mongoose = require('mongoose');
+const PostModel = require('../models/postSchema');
+const config = require('../config');
+const _ = require('lodash');
+
+const User = mongoose.model('User');
+
+exports.me = (req, res) => {
+  if (!req.headers.authorization) {
+    return res.status(401).end();
+  }
+
+  const token = req.headers.authorization.split(' ')[1];
+
+  return jwt.verify(token, config.jwtSecret, (err, decoded) => {
+    if (err) {
+      return res.status(401).end();
+    }
+    const userId = decoded.sub;
+    return res.status(200).send({ userId });
+  });
+};
+
+exports.getAllPost = (req, res) => {
+  PostModel.find({ }, (err, docs) => {
+    if (err) res.status(200).json({ message: 'database error' });
+    else res.status(200).json(docs);
+  });
+};
+
+exports.getPost = (req, res) => {
+  PostModel.findById({ _id: req.params.id }, (err, docs) => {
+    if (err || !docs) res.status(200).json({ message: 'database error' });
+    else { res.status(200).json(docs); }
+  });
+};
+
```

Comment:
if you are keeping a common post api it makes no sense to  name the controllers "userPostControllers';  
One solution using a common post api cd be to keep 3 layers of controller
- commonPost [ get all post etc]
-user specific
-admin specific

Reply:
I agree this name is not sensible, right now the api we have is admin rights are superset of user rights, so what I can do is change its name following the `routes/` folder names.